### ISSUE TITLE
transformers version 

### ISSUE THREAD

**Issue by wangjian026** [HLTCHKUST/QFS#1]
hello，i use the transformers(v3.0.2),and got multiple error "unresolved reference" in modeling_bart.py. if i use higher version of transformers(v3.1.0), the above error disappears，However,  at running time，since there are great differences between the two versions， there is something wrong occur in function "prepare_inputs_for_generation()" in modeling_bart.py.

**Comment by elevenofji** [comment]
not only the "prepare_inputs_for_generation", in lighting_base.py, "Adafactor" and "get_polynomial_decay_schedule_with_warmup" **cannot** be imported correctly.I'm just trying to find a solution

**Comment by wangjian026** [comment]
use transformers(v3.2.0), and change the code"encoder_outputs, decoder_past_key_values = past" of function"prepare_inputs_for_generation" in lighting_base.py to "encoder_outputs, decoder_past_key_values = (kwargs['encoder_outputs'],None)" may solve this problem.

**Comment by elevenofji** [comment]
> use transformers(v3.2.0), and change the code"encoder_outputs, decoder_past_key_values = past" of function"prepare_inputs_for_generation" in lighting_base.py to "encoder_outputs, decoder_past_key_values = (kwargs['encoder_outputs'],None)" may solve this problem.

GOOD idea. And I think sth about decoder_past_key_values, such as "decoder_past_key_values = outputs.decoder_past_key_values," in modeling_bart should be removed too.

BTW, have you tried the model with answer_relevance_score? I just found that it depends on another paper and there's a ckpt too. (https://github.com/HLTCHKUST/HLTC-MRQA)

**Comment by Iamfinethanksu** [comment]
Thank you for the discussion. Actually, our transformer's version for the experiment is a little bit twisted between the two version, for some historical issue. You may choose v3.0.2 and modify some function (input, or output ) accordingly, then it should be fine.

**Comment by elevenofji** [comment]
@Iamfinethanksu Yeah，u r right. The transformer's version has to be 3.0.2, and there're so many differences in the "input" and "output" between 3.0.2 and 3.2.0. I cannot get an expected result in 3.2.0 (even cannot get a normal decoding sentence, the reason may be that @wangjian026 solution may ignore the "decoder_past_key_values" directly. ). After fixing the transformer's version to 3.0.2 and changing some funcs, it works. I'll push the changed code to my repos.


Here are 10 hours fine-tuning results (without mrc's score).
<img width="632" alt="1638170521(1)" src="https://user-images.githubusercontent.com/32663690/143824770-465d6b07-70dc-458a-a9db-0e828fdaf1ef.png">

Here are paper's res:
![image](https://user-images.githubusercontent.com/32663690/143825202-6d6a7782-2218-42ac-a588-07795aa84e01.png)

The reason may be that the model takes "rouge2" as a val-metric.

@Iamfinethanksu Maybe you can emphasize the problem in README.md. (Also, the paper pdf link in README.md is wrong, linking to your another paper in emnlp19 XD).
Thx for @Iamfinethanksu and @wangjian026 's help.

**Comment by wangjian026** [comment]
> @Iamfinethanksu Yeah，u r right. The transformer's version has to be 3.0.2, and there're so many differences in the "input" and "output" between 3.0.2 and 3.2.0. I cannot get an expected result in 3.2.0 (even cannot get a normal decoding sentence, the reason may be that @wangjian026 solution may ignore the "decoder_past_key_values" directly. ). After fixing the transformer's version to 3.0.2 and changing some funcs, it works. I'll push the changed code to my repos.
> 
> Here are 10 hours fine-tuning results (without mrc's score). <img alt="1638170521(1)" width="632" src="https://user-images.githubusercontent.com/32663690/143824770-465d6b07-70dc-458a-a9db-0e828fdaf1ef.png">
> 
> Here are paper's res: ![image](https://user-images.githubusercontent.com/32663690/143825202-6d6a7782-2218-42ac-a588-07795aa84e01.png)
> 
> The reason may be that the model takes "rouge2" as a val-metric.
> 
> @Iamfinethanksu Maybe you can emphasize the problem in README.md. (Also, the paper pdf link in README.md is wrong, linking to your another paper in emnlp19 XD). Thx for @Iamfinethanksu and @wangjian026 's help.

GOOD idea！I have been trying to get a better result, but I failed. Could you send me the modified codes with  transformer's version 3.0.2 ? Thanks again.Thanks again, my email: wangjian026@126.com

**Comment by elevenofji** [comment]
> > @Iamfinethanksu Yeah，u r right. The transformer's version has to be 3.0.2, and there're so many differences in the "input" and "output" between 3.0.2 and 3.2.0. I cannot get an expected result in 3.2.0 (even cannot get a normal decoding sentence, the reason may be that @wangjian026 solution may ignore the "decoder_past_key_values" directly. ). After fixing the transformer's version to 3.0.2 and changing some funcs, it works. I'll push the changed code to my repos.
> > Here are 10 hours fine-tuning results (without mrc's score). <img alt="1638170521(1)" width="632" src="https://user-images.githubusercontent.com/32663690/143824770-465d6b07-70dc-458a-a9db-0e828fdaf1ef.png">
> > Here are paper's res: ![image](https://user-images.githubusercontent.com/32663690/143825202-6d6a7782-2218-42ac-a588-07795aa84e01.png)
> > The reason may be that the model takes "rouge2" as a val-metric.
> > @Iamfinethanksu Maybe you can emphasize the problem in README.md. (Also, the paper pdf link in README.md is wrong, linking to your another paper in emnlp19 XD). Thx for @Iamfinethanksu and @wangjian026 's help.
> 
> GOOD idea！I have been trying to get a better result, but I failed. Could you send me the modified codes with transformer's version 3.0.2 ? Thanks again.Thanks again, my email: [wangjian026@126.com](mailto:wangjian026@126.com)

you can just download in https://github.com/elevenofji/bart_test.

**Comment by wangjian026** [comment]
> > > @Iamfinethanksu Yeah，u r right. The transformer's version has to be 3.0.2, and there're so many differences in the "input" and "output" between 3.0.2 and 3.2.0. I cannot get an expected result in 3.2.0 (even cannot get a normal decoding sentence, the reason may be that @wangjian026 solution may ignore the "decoder_past_key_values" directly. ). After fixing the transformer's version to 3.0.2 and changing some funcs, it works. I'll push the changed code to my repos.
> > > Here are 10 hours fine-tuning results (without mrc's score). <img alt="1638170521(1)" width="632" src="https://user-images.githubusercontent.com/32663690/143824770-465d6b07-70dc-458a-a9db-0e828fdaf1ef.png">
> > > Here are paper's res: ![image](https://user-images.githubusercontent.com/32663690/143825202-6d6a7782-2218-42ac-a588-07795aa84e01.png)
> > > The reason may be that the model takes "rouge2" as a val-metric.
> > > @Iamfinethanksu Maybe you can emphasize the problem in README.md. (Also, the paper pdf link in README.md is wrong, linking to your another paper in emnlp19 XD). Thx for @Iamfinethanksu and @wangjian026 's help.
> > 
> > 
> > GOOD idea！I have been trying to get a better result, but I failed. Could you send me the modified codes with transformer's version 3.0.2 ? Thanks again.Thanks again, my email: [wangjian026@126.com](mailto:wangjian026@126.com)
> 
> you can just download in https://github.com/elevenofji/bart_test.

Thank you very much for your work, I have downloaded the code, however, a file named modeling_bart seems to have been lost in the src file

**Comment by elevenofji** [comment]
> > > > @Iamfinethanksu Yeah，u r right. The transformer's version has to be 3.0.2, and there're so many differences in the "input" and "output" between 3.0.2 and 3.2.0. I cannot get an expected result in 3.2.0 (even cannot get a normal decoding sentence, the reason may be that @wangjian026 solution may ignore the "decoder_past_key_values" directly. ). After fixing the transformer's version to 3.0.2 and changing some funcs, it works. I'll push the changed code to my repos.
> > > > Here are 10 hours fine-tuning results (without mrc's score). <img alt="1638170521(1)" width="632" src="https://user-images.githubusercontent.com/32663690/143824770-465d6b07-70dc-458a-a9db-0e828fdaf1ef.png">
> > > > Here are paper's res: ![image](https://user-images.githubusercontent.com/32663690/143825202-6d6a7782-2218-42ac-a588-07795aa84e01.png)
> > > > The reason may be that the model takes "rouge2" as a val-metric.
> > > > @Iamfinethanksu Maybe you can emphasize the problem in README.md. (Also, the paper pdf link in README.md is wrong, linking to your another paper in emnlp19 XD). Thx for @Iamfinethanksu and @wangjian026 's help.
> > > 
> > > 
> > > GOOD idea！I have been trying to get a better result, but I failed. Could you send me the modified codes with transformer's version 3.0.2 ? Thanks again.Thanks again, my email: [wangjian026@126.com](mailto:wangjian026@126.com)
> > 
> > 
> > you can just download in https://github.com/elevenofji/bart_test.
> 
> Thank you very much for your work, I have downloaded the code, however, a file named modeling_bart seems to have been lost in the src file

Sorry for that. I've fixed. Maybe we can discuss in https://github.com/elevenofji/bart_test. This issue can be closed.
